File: OdonSys-SPA/src/app/core/models/users/api/register-user-request.ts
import { Country } from '../../../../core/enums/country.enum';

export class RegisterUserRequest {
  constructor(
    public name: string,
    public surname: string,
    public document: string,
    public password: string,
    public phone: string,
    public email: string,
    public country: Country,
    public secondSurname?: string,
    public middleName?: string,
    public userId?: string,
    public userName?: string
  ) {}
}

File: OdonSysBackEnd/Resource/Contract/Contract.Administration/Users/RegisterUserRequest.cs
using Utilities.Enums;

namespace Contract.Administration.Users;

public record RegisterUserRequest
(
    string Name,
    string MiddleName,
    string Surname,
    string SecondSurname,
    string Document,
    string Password,
    string Phone,
    string Email,
    Country Country
);
